Add `/api/teams` endpoint

isamrish opened the pull request:
Signed-off-by: Amrish Kushwaha <askmaurya48@gmail.com>

This commit will add functionality that
/api/reviewers will list teams when no query parameter is passed

isamrish committed:
List teams of the org when no parameter is passed

Signed-off-by: Amrish Kushwaha <askmaurya48@gmail.com>

This commit will add functionality that
/api/reviewers will list teams when no query parameter is passed

itaditya commented:
@IsAmrish glad you picked this task, the logic seems right to me. I'll check it locally once I come back from home.
Things you need to change-

1. Remove the formatting you have done. I use prettier for formatting in this project. (single quote, trailing comma)
2. Instead of if-else, do an early return like this

```js
if(team.length === 0) {
  // team fetch logic
  return;
}

// old logic for when team length is not 0
```

isamrish committed:
refactor code and add .prettierrc file

Signed-off-by: Amrish Kushwaha <askmaurya48@gmail.com>

This commit will
- refactor logic for /api/reviewers end point when not parameter is
passed. Instead of using if else, early return is used.
- Add .prettierrc to properly format the server.js file with single
quotes and trailing commas

isamrish commented:
@itaditya  -- I have added .prettierrc file to format and refactored the code according to your suggestions.

itaditya commented on .prettierrc:
```diff
@@ -0,0 +1,4 @@
+{
+  "trailingComma": "es5",
+  "singleQuote": true
```
add `printWidth` equal to `120` also and format accordingly

itaditya commented:
@IsAmrish I think it would be better if you make a separate API for getting list of all teams in an ORG. Then on frontend based on query param provided or not you fetch this API and show dropdown.

It's great you abstracted the get teams logic in a function so it would be very easy to make the change.

Make this change and the prettier comment I have left and I'll merge the PR

isamrish commented:
@itaditya  - separating API for teams is good idea. I think `teams` is good end point.
It also means I need to go back to previous code for the `reviewers` endpoint. right!!

isamrish committed:
Separate API for getting teams and Add printWidth for format

Signed-off-by: Amrish Kushwaha <askmaurya48@gmail.com>

This commit will
- Seperate the logic for teams endpoint from reviewers end point
- Add printWidth key and set its value to 120 for correct formating

itaditya commented:
@IsAmrish There is one bug in your code. You are sending `name` of team. Actually name can be 'My Team' but slug would be `my-team` and every further GitHub API uses the team slug to get that team's data.

The `name` field can be used in the dropdown as the label. That means in dropdown user will see the list of names of teams. When they click on one item, we will find the slug corresponding to that item and then go ahead.

So basically we need both `name` and `slug` from API, so API res should be->

```js
{
  "teamsList": [
    {
      "name": "Some Team",
      "slug": "someteam"
    },
    {
      "name": "My Team",
      "slug": "my-team"
    }
  ]
}

```

isamrish commented:
@itaditya -- thanks for pointing out. Fixing it.

isamrish committed:
Return name and slug of teams on /api/teams end point

Signed-off-by: Amrish Kushwaha <askmaurya48@gmail.com>

Instead of returning array of team name, return an object of
name and slug

itaditya commented:
@IsAmrish thanks for working on it. Congrats your PR is merged!

![Congrats](https://media.giphy.com/media/sBLcw5Ic4QUTK/giphy.gif)
